src/server/scripts/Northrend/ChamberOfAspects/ObsidianSanctum/boss_sartharion.cpp

```diff
@@ -673,6 +674,17 @@ class boss_sartharion : public CreatureScript
             dragonsCount = 0;
         }
 
+        bool IsTargetInBounds(Unit const* victim) const
+        {
+            if (!victim || victim->GetPositionX() < SartharionBoundary[0] || victim->GetPositionX() > SartharionBoundary[1] || victim->GetPositionY() < SartharionBoundary[2] || victim->GetPositionY() > SartharionBoundary[3])
+            {
+                return false;
+            }
+
+            return true;
+
+        }
```

Comment:
```suggestion
        bool IsTargetInBounds(Unit const* victim) const
        {
            return !(!victim ||
            victim->GetPositionX() < SartharionBoundary[0] ||
            victim->GetPositionX() > SartharionBoundary[1] ||
            victim->GetPositionY() < SartharionBoundary[2] ||
            victim->GetPositionY() > SartharionBoundary[3])
        }
```

Reply:
in further code i added exception for big island, this code would work for older version  and wont work for that exception but anyways thanks